src/app/ReadClient.cpp

```diff
@@ -189,9 +208,25 @@ CHIP_ERROR ReadClient::SendStatusReport(CHIP_ERROR aError)
     System::PacketBufferHandle msgBuf = buf.Finalize();
     VerifyOrReturnLogError(!msgBuf.IsNull(), CHIP_ERROR_NO_MEMORY);
 
+    if (IsSubscription())
```

Comment:
currently, for read, we always send status report. since spec mentioned, "Data reports (i.e. non-empty ReportData messages) SHALL always require a response. In those cases, the SuppressResponse flag isn’t applicable and SHALL not be sent." in https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/data_model/Encoding-Specification.adoc#reportdatamessage